sklearn/feature_selection/_from_model.py

```diff
@@ -237,17 +239,30 @@ def __init__(
         self.max_features = max_features
 
     def _get_support_mask(self):
-        # SelectFromModel can directly call on transform.
+        estimator = getattr(self, "estimator_", self.estimator)
+        max_features = getattr(self, "max_features_", self.max_features)
+
         if self.prefit:
-            estimator = self.estimator
-        elif hasattr(self, "estimator_"):
-            estimator = self.estimator_
-        else:
+            try:
+                check_is_fitted(self.estimator)
+            except NotFittedError as exc:
+                raise NotFittedError(
+                    "When `prefit=True`, `estimator` is expected to be a fitted "
+                    "estimator."
+                ) from exc
+        if callable(max_features):
```

Comment:
It's not immediately clear that it corresponds to having called transform without having fitted first. Maybe a comment would help